index.js

```diff
@@ -33,28 +33,61 @@ function TinySDF(fontSize, buffer, radius, cutoff, fontFamily, fontWeight) {
     this.middle = Math.round((size / 2) * (navigator.userAgent.indexOf('Gecko/') >= 0 ? 1.2 : 1));
 }
 
-TinySDF.prototype.draw = function (char) {
+TinySDF.prototype._draw = function (char, getMetrics) {
     this.ctx.clearRect(0, 0, this.size, this.size);
     this.ctx.fillText(char, this.buffer, this.middle);
+    var textMetrics = this.ctx.measureText(char);
+    // Older browsers only expose the glyph width
+    // This is enough for basic layout with all glyphs using the same fixed size
+    var metrics = {
+      width: this.size,
+      height: this.size,
+      left: 0,
+      top: -this.size / 3,
+      advance: textMetrics.width
+    };
+
+    var imgData;
+    // If the browser supports bounding box metrics, we can generate a smaller
+    // SDF. This is a significant performance win.
+    if (getMetrics && textMetrics.actualBoundingBoxLeft !== undefined) {
+      metrics.width = Math.ceil(textMetrics.actualBoundingBoxRight - textMetrics.actualBoundingBoxLeft);
+      metrics.height = Math.ceil(textMetrics.actualBoundingBoxAscent + textMetrics.actualBoundingBoxDescent);
+      metrics.top = Math.floor(textMetrics.actualBoundingBoxAscent) - this.middle;
+
+      // The integer/pixel part of the top alignment is encoded in metrics.top
+      // The remainder is implicitly encoded in the rasterization
+      var imgTop = this.middle - Math.ceil(textMetrics.actualBoundingBoxAscent) - this.buffer;
+      imgData = this.ctx.getImageData(0, imgTop, metrics.width + 2 * this.buffer, metrics.height + 2 * this.buffer);
```

Comment:
Should we trim the SDF even if the bbox isn't available, at least by width measured earlier?

Reply:
The width measured earlier is the layout width (or "advance"). It's possible for the rasterization of the glyph to extend past that (for instance in an italic font the bounding boxes would almost all be wider than the advances).